Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandStoreBuyItem.java

```diff
@@ -83,6 +95,36 @@ public void run(Map<String,String> parameters) throws UserErrorMessage {
 		ds.beginTransaction();
 		try
 		{
+			// If quantity item, need to refetch in transaction, to make 
+			// sure enough quantity available to purchase.
+			if(quantity > 1)
+			{
+				item = ds.refetch(item);

```

Comment:
Any entity fetched outside of the transaction must be refetched inside of the transaction. Refetching guarantees that it reads from the DB instead of memcache and does so immediately.